src/vm/gcenv.os.cpp

```diff
@@ -42,6 +66,20 @@ bool GCToOSInterface::Initialize()
     g_pageSizeUnixInl = GetOsPageSize();
 #endif
 
+    uintptr_t pmask, smask;
+    if (!!::GetProcessAffinityMask(::GetCurrentProcess(), (PDWORD_PTR)&pmask, (PDWORD_PTR)&smask))
```

Comment:
>GetProcessAffinityMask [](start = 12, length = 22)

I must be missing something - if you are running in cpu groups this would make return pmask/smask as 0. so g_processAffinitySet will be 0. is there code that fills it in for the cpu group case on windows?

Reply:
This is not changing the current behavior. This process mask getting has just moved from GCHeap::Initialize to GCToOSInterface::Initialize.
As for the mask, it never returns 0 at the initialization time as the process is running in a single CPU group and the affinity mask is relative to that group. 

Actually, the GC code even without my changes ignores the process affinity mask when CPU groups are enabled. See the `process_mask` that was replaced by `g_processAffinitySet`. 
I guess the reason was that the process affinity mask at launch time (that can be specified using the `AFFINITY` parameter of the `start` command) is always relative to a single group, but we use more groups when groups are enabled and thus that mask doesn't make much sense (well, we could say that we would apply that mask on the original group the process was launched in and the rest of the groups are not affected, but that would be weird).

So I'd suggest modifying the behavior for the case when CPU groups are enabled. We would completely ignore the launch-time process affinity mask (which we already do), but we would use the mask or ranges specified by the config knobs as-is to set the `g_processAffinitySet`. In case the user didn't set any of these knobs, the `g_processAffinitySet` would be set to contain all the processors on the system.

Reply:
we talked about this - I was mistakenly thinking that the CPU group affinitizing happened before GC init which cause my confusion; and correct, the GCHeapAffinitizeRanges should be used in all cases that are applicable, including on Windows where it's running in multiple CPU groups.